service/internal/proctelemetry/process_telemetry.go

```diff
@@ -31,100 +48,224 @@ type processMetrics struct {
 	ballastSizeBytes  uint64
 	proc              *process.Process
 
-	processUptime *metric.Float64DerivedCumulative
-	allocMem      *metric.Int64DerivedGauge
-	totalAllocMem *metric.Int64DerivedCumulative
-	sysMem        *metric.Int64DerivedGauge
-	cpuSeconds    *metric.Float64DerivedCumulative
-	rssMemory     *metric.Int64DerivedGauge
+	processUptime *ocmetric.Float64DerivedCumulative
+	allocMem      *ocmetric.Int64DerivedGauge
+	totalAllocMem *ocmetric.Int64DerivedCumulative
+	sysMem        *ocmetric.Int64DerivedGauge
+	cpuSeconds    *ocmetric.Float64DerivedCumulative
+	rssMemory     *ocmetric.Int64DerivedGauge
+
+	// otel metrics
+	otelProcessUptime asyncfloat64.Counter
+	otelAllocMem      asyncint64.Counter
```

Comment:
we should keep the same metric types, can't change allocMem from gauge to counter as it yields wrong behavior and is not compatible.

Reply:
Thanks for pointing this out changed the memory instruments to gauges!